### PR TITLE
e2e: ROOT_SECRET from .env file

### LINKED ISSUE #633: e2e-tests: Add ROOT_SECRET environment variable
- [x] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

Currently the passwort of the root user (ROOT_SECRET) is hardcoded and cannot be changed. Adding an env variable to set a custom ROOT_SECRET  would prevent tester issues while setting up the e2e-test locally.

### PR DESCRIPTION
**Pull request by galethil**
### Checklist

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/openkfw/TruBudget/blob/master/CONTRIBUTING.md).

### Description

ROOT_SECRET in e2e tests in now configurable using .env file. Just create .env file from  .env.example file. 
You can change the value in the file if you need to.
Variable must start with CYPRESS_ for cypress to be able to read it.

Closes #633

### CODE REVIEW

File: e2e-test/cypress/plugins/index.js [hunk 1]
```diff
@@ -115,4 +116,6 @@ module.exports = (on, _config) => {
       return options;
     }
   });
+  _config = dotenvPlugin(_config);
```
Comment (Stezido) [inline e2e-test/cypress/plugins/index.js:119]: Rename the _config  to config since it's used now.

File: e2e-test/.env.example [hunk 2]
```diff
@@ -0,0 +1 @@
+CYPRESS_ROOT_SECRET=root-secret
```
Comment (Stezido) [inline e2e-test/.env.example:1]: I think we can use the ROOT_SECRET itself. I cannot think of a scenario where the rootsecret of cypress is a different one than the one in the api or elsewhere

**Review by Stezido — CHANGES_REQUESTED** [review]
The documentation has to be updated, but since there is another team meber working on a refactoring of the current contributor documentation you can skip it

File: e2e-test/cypress/plugins/index.js (hunk 1 above)
Comment (galethil) [reply e2e-test/cypress/plugins/index.js]: updated.

File: e2e-test/.env.example (hunk 2 above)
Comment (galethil) [reply e2e-test/.env.example:1]: @Stezido I'm not sure if that is possible since cypress is consuming only env variables starting with `CYPRESS_`

Comment (galethil) [reply e2e-test/.env.example:1]: https://docs.cypress.io/guides/guides/environment-variables.html#Option-3-CYPRESS

Comment (Stezido) [reply e2e-test/.env.example:1]: true, thanks for the information link

**Review by Stezido — DISMISSED** [review]

**Review by Stezido — COMMENTED** [review]
Please check again if the docker-compose for the e2e-tests is working:
```
sh scripts/testing/start-e2e-tests.sh 
```

Note that the docker-compose e2etest process can be improved by implementing a way to check if it's already provisioned, but it should work anyway

File: doc/installation/Developer-Setup.md [hunk 3]
```diff
@@ -463,11 +465,11 @@ In the `e2e-test` folder you can run the following commands:
 You need to run cypress while also specifying the urls of the frontend, api and excel export service you are using
 
 ```bash
-npm run cypress -- --config baseUrl=http://localhost:3000 --env API_BASE_URL=http://localhost:8080,EXPORT_SERVICE_BASE_URL=http://localhost:8888
+npm run cypress -- --config baseUrl=http://localhost:3000 --env API_BASE_URL=http://localhost:8080,EXPORT_SERVICE_BASE_URL=http://localhost:8888,ROOT_SECRET=root-secret
```
Comment (m-janos) [inline doc/installation/Developer-Setup.md:468]: @galethil when you are using .env file, there is no need to use --env flag. The .env file has precedence. There should be mention of two possible runs: One with .env file and the second with --env flag to not cause confusion.

Comment (galethil) [reply doc/installation/Developer-Setup.md:468]: @Stezido what is the preferred way of running e2e tests? What should be put to Developer documentation?

Comment (Stezido) [reply doc/installation/Developer-Setup.md:480]: @m-janos yes true but we have to make sure the executor is in the right folder
@galethil If all e2e-test should be executed and only the end result is important the all-in-one-docker-compose should be used. If the docker-compose is already running of course only the required services and the e2e-tests can be restarted. If I only want to test a specific test I would use the ui with npm run cypress or the script with npm run e2e-test:here. We should definitely document the prefered ways for both scenarios:
1. How to run e2e-tests while I can do other stuff and only want the end result.
2. How to run only a set of tests so the try and error cycle is as fast as possible

Comment (Stezido) [reply doc/installation/Developer-Setup.md:480]: I will only add the all-in-one docker-compose documentation in this PR. The other changes could be done in  https://github.com/openkfw/TruBudget/issues/769

**Comment by m-janos** [comment]
@galethil  After running `sh scripts/testing/start-e2e-tests.sh` Cypress cannot connect to the API.

API container log:
```
Problem with establishing connection to min.io and creating bucket.
Error during searching for bucket Error: getaddrinfo ENOTFOUND nginx
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26) {
  errno: 'ENOTFOUND',
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'nginx'
}
```

**Comment by Stezido** [comment]
@m-janos ty for noticing I added minio to the compose file should work now

**Review by m-janos — DISMISSED** [review]

File: e2e-test/cypress/integration/subproject_permissions_spec.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const testUser2 = { id: "jxavier", displayname: "Jane Xavier", password: "test"
 const testGroup = { id: "admins", displayname: "Admins" };
 let projectId, subprojectId, permissionsBeforeTesting, baseUrl, apiRoute;
 const subprojectDisplayname = "subproject assign test";
-const rootSecret = "root-secret";
+const rootSecret = Cypress.env("ROOT_SECRET");
 
 describe("Subproject Permissions", function() {
```
Comment (m-janos) [inline e2e-test/cypress/integration/subproject_permissions_spec.js:11]: One test is failing (locally only) with following error:
```
  1) Subproject Permissions
       Confirmation of multiple grant permission changes grants permissions correctly:
     CypressError: `cy.wait()` timed out waiting `5000ms` for the 3rd request to the route: `listSubprojectPermissions`. No request ever occurred.
```

**Review by m-janos — APPROVED** [review]

**Pull request merged**